configuration/src/main/java/io/airlift/configuration/AbstractConfigurationAwareModule.java

```diff
@@ -79,6 +79,17 @@ protected synchronized <T> T buildConfigObject(Class<T> configClass, String pref
         return configurationFactory.build(configClass, prefix);
     }
 
+    protected synchronized <T> T buildConfigObject(Key<T> key, Class<T> configClass)
```

Comment:
Binding config objects is actually unrelated to conditional module. I think we should go the opposite way and remove the `bindConfig` from this code.